### PR TITLE
Backward compatibility in DrILL file loading

### PR DESCRIPTION
**Pull request by gui-co**
**Description of work.**

With this PR, the DrILL interface can load new and old rundex files.
#29782 changed the way the rundex files are exported and imported. This PR introduce a backward compatibility.

**To test:**

* copy the two following json in separate files
* open the DrILL interface
* observe that both files load fine and produce the same table in DrILL

<details><summary>oldRundex.mrd</summary><p>

```json
{
    "Instrument": "D11",
    "AcquisitionMode": "SANS",
    "VisualSettings": {
        "FoldedColumns": [
            "SampleTransmissionRuns",
            "AbsorberRuns",
            "BeamRuns",
            "FluxRuns",
            "ContainerRuns",
            "ContainerTransmissionRuns",
            "TransmissionAbsorberRuns",
            "TransmissionBeamRuns",
            "MaskFiles",
            "ReferenceFiles",
            "OutputWorkspace"
        ],
        "HiddenColumns": [],
        "ColumnsOrder": [
            "SampleRuns",
            "SampleTransmissionRuns",
            "AbsorberRuns",
            "BeamRuns",
            "FluxRuns",
            "ContainerRuns",
            "ContainerTransmissionRuns",
            "TransmissionAbsorberRuns",
            "TransmissionBeamRuns",
            "MaskFiles",
            "ReferenceFiles",
            "OutputWorkspace",
            "SampleThickness",
            "CustomOptions"
        ]
    },
    "GlobalSettings": {
        "ThetaDependent": true,
        "SensitivityMaps": "",
        "DefaultMaskFile": "",
        "NormaliseBy": "Timer",
        "SampleThickness": 0.1,
        "BeamRadius": [
            0.1
        ],
        "TransmissionBeamRadius": 0.1,
        "WaterCrossSection": 1.0,
        "OutputType": "I(Q)",
        "CalculateResolution": "None",
        "DefaultQBinning": "PixelSizeBased",
        "BinningFactor": 1.0,
        "OutputBinning": [],
        "NPixelDivision": 1,
        "NumberOfWedges": 0,
        "WedgeAngle": 30.0,
        "WedgeOffset": 0.0,
        "AsymmetricWedges": false,
        "MaxQxy": [
            -1.0
        ],
        "DeltaQ": [
            -1.0
        ],
        "IQxQyLogBinning": false,
        "OutputPanels": false
    },
    "Samples": [
        {
            "SampleRuns": "2889",
            "CustomOptions": {
                "ThetaDependent": true
            }
        }
    ]
}
```
</p></details>

<details><summary>newRundex.mrd</summary><p>

```json
{
    "Instrument": "D11",
    "AcquisitionMode": "SANS",
    "VisualSettings": {
        "FoldedColumns": [
            "SampleTransmissionRuns",
            "AbsorberRuns",
            "BeamRuns",
            "FluxRuns",
            "ContainerRuns",
            "ContainerTransmissionRuns",
            "TransmissionAbsorberRuns",
            "TransmissionBeamRuns",
            "MaskFiles",
            "ReferenceFiles",
            "OutputWorkspace"
        ],
        "HiddenColumns": [],
        "ColumnsOrder": [
            "SampleRuns",
            "SampleTransmissionRuns",
            "AbsorberRuns",
            "BeamRuns",
            "FluxRuns",
            "ContainerRuns",
            "ContainerTransmissionRuns",
            "TransmissionAbsorberRuns",
            "TransmissionBeamRuns",
            "MaskFiles",
            "ReferenceFiles",
            "OutputWorkspace",
            "SampleThickness",
            "CustomOptions"
        ]
    },
    "GlobalSettings": {
        "ThetaDependent": true,
        "SensitivityMaps": "",
        "DefaultMaskFile": "",
        "NormaliseBy": "Timer",
        "SampleThickness": 0.1,
        "BeamRadius": [
            0.1
        ],
        "TransmissionBeamRadius": 0.1,
        "WaterCrossSection": 1.0,
        "OutputType": "I(Q)",
        "CalculateResolution": "None",
        "DefaultQBinning": "PixelSizeBased",
        "BinningFactor": 1.0,
        "OutputBinning": [],
        "NPixelDivision": 1,
        "NumberOfWedges": 0,
        "WedgeAngle": 30.0,
        "WedgeOffset": 0.0,
        "AsymmetricWedges": false,
        "MaxQxy": [
            -1.0
        ],
        "DeltaQ": [
            -1.0
        ],
        "IQxQyLogBinning": false,
        "OutputPanels": false
    },
    "Samples": [
        {
            "SampleRuns": "2889",
            "ThetaDependent": true
        }
    ]
}
```
</p></details>

*This does not require release notes*, internal changes only.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Interface/ui/drill/test/DrillSampleTest.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+class DrillSampleTest(unittest.TestCase):
+
+    def setUp(self):
+        self.sample = DrillSample()
+
+    def test_init(self):
+        self.assertDictEqual(self.sample._parameters, {})
+
+    def test_setParameters(self):
```
Comment (gvardany) [inline scripts/Interface/ui/drill/test/DrillSampleTest.py:21]: The only issue is that old/new is binary. In theory we might have many versions of rundex format, so you should rather organize separate tests for v1, v2, etc.

**Review by gvardany — COMMENTED** [review]
I could load both into the interface. I just wonder whether RundexIOTest might be a better place for the tests with difference rundex versions.

**Comment by gui-co** [comment]
That is what I was thinking when reading your comment about the different versions of Rundex files. I will change that.

**Review by gvardany — APPROVED** [review]
Looks good.

**Pull request merged**